lib/plugins/leetcode.js

```diff
@@ -552,11 +552,18 @@ function parseCookie(cookie, cb) {
   };
 }
 
-function saveAndGetUser(user, cb, cookieData) {
-  user.sessionId = cookieData.sessionId;
-  user.sessionCSRF = cookieData.sessionCSRF;
-  session.saveUser(user);
-  plugin.getUser(user, cb);
+function requestLeetcodeAndSave(request, leetcodeUrl, user, cb, party) {
+  request.get({url: leetcodeUrl}, function(e, resp, body) {
+    const redirectUri = resp.request.uri.href;
+    if (redirectUri !== 'https://leetcode.com/') {
+      return cb(`${party} login failed or ${party} did not link to LeetCode`);
```

Comment:
One suggestion here, since the LeetCode official website is using verb `connect`, I suggest we use replace `link` with `connect`.